suitcase/tiff/__init__.py

```diff
@@ -1,25 +1,325 @@
-from ._version import get_versions
-__version__ = get_versions()['version']
-del get_versions
-
-
 # Suitcase subpackages must follow strict naming and interface conventions. The
 # public API should include some subset of the following. Any functions not
 # implemented should be omitted, rather than included and made to raise
 # NotImplementError, so that a client importing this library can immediately
 # know which portions of the suitcase API it supports without calling any
 # functions.
 #
-# def export(...)
-#     ...
-#
-#
-# def ingest(...):
-#     ...
-#
-#
-# def reflect(...):
-#     ...
-#
-#
-# handlers = []
+from collections import defaultdict
+from pathlib import Path
+import json
+import tifffile
+import event_model
+import numpy
+import suitcase.utils
+from ._version import get_versions
+
+__version__ = get_versions()['version']
+del get_versions
+
+
+class SuitcaseTiffError(Exception):
+    ...
+
+
+class NonSupportedDataShape(SuitcaseTiffError):
+    '''used to indicate that non-supported data shape is being saved
+    '''
+    ...
+
+
+def export(gen, directory, file_prefix='{uid}-', **kwargs):
+    """
+    Export a stream of documents to TIFF stack(s) and one JSON file.
+
+    This creates one file named ``<filepath>_meta.json`` and a file named
```

Comment:
filepath is not defined here would it be better to use ,``<directory>\<file_prefix>....``?

Reply:
Can we use a UNIX/OSX-land forward slash rather than Windows-land backslash? In fact, `Path` will use whichever based on what machine you are on, but I expect we will generally be exporting from Linux.

Reply:
change made as requested